src/wp-includes/class-wp-block-bindings-source.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Block Bindings API: WP_Block_Bindings_Source class.
+ *
+ *
+ * @package WordPress
+ * @subpackage Block Bindings
+ * @since 6.5.0
+ */
+
+/**
+ * Class representing block bindings source.
+ *
+ * This class is designed for internal use by the Block Bindings registry.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @see WP_Block_Bindings_Registry
+ */
+final class WP_Block_Bindings_Source {
+
+	public function __construct( string $name, array $source_properties ) {
+		$this->name = $name;
+
+		/* Validate that the source properties contain the label */
+		if ( ! isset( $source_properties['label'] ) ) {
```

Comment:
Can the validation live in one place? The registry also has some validation present:

https://github.com/WordPress/wordpress-develop/blob/a770c6da8dee44a060427fa82aac7acf25ff5a44/src/wp-includes/class-wp-block-bindings-registry.php#L67-L93

Is `_doing_it_wrong` able to prevent the registration when the `get_value_callback` is not provided? I'm wondering which place is better for such logic.

Reply:
I moved the validation to the registry now and also added a couple tests to check if:
- the label is passed
- the `'get_value_callback'` is passed
- the `'get_value_callback'` is a `callable`